code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityTests.java

```diff
@@ -269,6 +270,186 @@ public void call(Object o) {}
 		assertEquals(AdobeError.UNEXPECTED_ERROR, errorCapture.get(KEY_CAPTUREDERRORCALLBACK));
 	}
 
+	// ========================================================================================
+	// getUrlVariables API
+	// ========================================================================================
+	@Test
+	public void testGetUrlVariables() {
+		// setup
+		final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+		final ArgumentCaptor<AdobeCallback> adobeCallbackCaptor = ArgumentCaptor.forClass(AdobeCallback.class);
+		final ArgumentCaptor<ExtensionErrorCallback> extensionErrorCallbackCaptor = ArgumentCaptor.forClass(
+			ExtensionErrorCallback.class
+		);
+		final List<String> callbackReturnValues = new ArrayList<>();
+
+		// test
+		Identity.getUrlVariables(
+			new AdobeCallback<String>() {
+				@Override
+				public void call(String s) {
+					callbackReturnValues.add(s);
+				}
+			}
+		);
+
+		// verify
+		PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
+		MobileCore.dispatchEventWithResponseCallback(
+			eventCaptor.capture(),
+			adobeCallbackCaptor.capture(),
+			extensionErrorCallbackCaptor.capture()
+		);
+
+		// verify the dispatched event details
+		Event dispatchedEvent = eventCaptor.getValue();
+		assertEquals(IdentityConstants.EventNames.IDENTITY_REQUEST_URL_VARIABLES, dispatchedEvent.getName());
+		assertEquals(IdentityConstants.EventType.EDGE_IDENTITY.toLowerCase(), dispatchedEvent.getType());
+		assertEquals(IdentityConstants.EventSource.REQUEST_IDENTITY.toLowerCase(), dispatchedEvent.getSource());
+		assertTrue(dispatchedEvent.getEventData().containsKey("urlvariables"));
+
+		// verify callback responses
+		Map<String, Object> urlVariablesResponse = new HashMap<>();
+		urlVariablesResponse.put("urlvariables", "test-url-variable-string");
+
+		adobeCallbackCaptor.getValue().call(buildUrlVariablesResponseEvent(urlVariablesResponse));
+		assertEquals("test-url-variable-string", callbackReturnValues.get(0));
+		// TODO - enable when ExtensionError creation is available
+		// should not crash on calling the callback
+		//extensionErrorCallback.error(ExtensionError.UNEXPECTED_ERROR);
+	}
+
+	@Test
+	public void testGetUrlVariables_nullCallback() {
+		// test
+		Identity.getExperienceCloudId(null);
```

Comment:
`Identity.getUrlVariables(null)` ?

Reply:
yes, I was working on these tests when you posted this :D